Add module for the switching endpoint of the API:

earendilfr opened the pull request:
Permit to close the following tasks:
- #72
- #73
- #74
- #75
- #76
- #77

WhaleJ84 commented:
Finished reviewing; works perfectly fine on my end. All the recommendations above are minor and I can do myself but I thought I'd note them down in case you want to resolve them yourself. Will merge in if no reply is given after 3 days. Thanks for adding!

earendilfr commented:
Ok: I have updated the PR with the recommendation provide by pylint.

Just one question: for your second remarks: did you prefer I replace 
```python
      def get_link(self, link: int):
      ¦   """
      ¦   Retrieves Link by ID
  
      ¦   :param link: Should be a link ID (integer)
      ¦   """
```
by 
```python
      def get_link(self, link: str):
      ¦   """
      ¦   Retrieves Link by ID
  
      ¦   :param link: Should be a link ID
      ¦   """
```

WhaleJ84 commented:
> Ok: I have updated the PR with the recommendation provide by pylint.
> 
> Just one question: for your second remarks: did you prefer I replace
> 
> ```python
>       def get_link(self, link: int):
>       ¦   """
>       ¦   Retrieves Link by ID
>   
>       ¦   :param link: Should be a link ID (integer)
>       ¦   """
> ```
> 
> by
> 
> ```python
>       def get_link(self, link: str):
>       ¦   """
>       ¦   Retrieves Link by ID
>   
>       ¦   :param link: Should be a link ID
>       ¦   """
> ```

I tested both strings and integers and they both work. In my experience, the LibreNMS API tends to prefer string interpretations of integers, so I would go with strings. You could always do a Union instead and accept both.

earendilfr commented:
> > Ok: I have updated the PR with the recommendation provide by pylint.
> > Just one question: for your second remarks: did you prefer I replace
> > ```python
> >       def get_link(self, link: int):
> >       ¦   """
> >       ¦   Retrieves Link by ID
> >   
> >       ¦   :param link: Should be a link ID (integer)
> >       ¦   """
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > by
> > ```python
> >       def get_link(self, link: str):
> >       ¦   """
> >       ¦   Retrieves Link by ID
> >   
> >       ¦   :param link: Should be a link ID
> >       ¦   """
> > ```
> 
> I tested both strings and integers and they both work. In my experience, the LibreNMS API tends to prefer string interpretations of integers, so I would go with strings. You could always do a Union instead and accept both.

OK: so, i have updated the PR and now use str as parameter.
